python/ray/tests/utils.py

```diff
@@ -94,3 +95,17 @@ def wait_for_errors(error_type, num_errors, timeout=10):
             return
         time.sleep(0.1)
     raise Exception("Timing out of wait.")
+
+
+def try_remove_directory(path):
+    if not os.path.exists(path):
+        return
+
+    for _ in range(3):
```

Comment:
Why do this 3 times?

Reply:
It is just a maximum retry. I found that sometimes the dir removal fails.

Reply:
Why does it fail? Could we do a `while(os.path.exists(...))` instead of a maximum retry?

Reply:
I have updated related code. It looks like we don't need it anymore.